src/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php

```diff
@@ -23,6 +23,15 @@ abstract class AbstractHtmlProcessor
      */
     const CONTENT_TYPE_META_TAG = '<meta http-equiv="Content-Type" content="text/html; charset=utf-8">';
 
+    /**
+     * @var string Regular expression part to match tag names that PHP's DOMDocument implementation is not aware are
+     *      self-closing. These are mostly HTML5 elements, but for completeness <command> (obsolete) and <keygen>
+     *      (deprecated) are also included.
+     *
+     * @see https://bugs.php.net/bug.php?id=73175
+     */
+    const PHP_UNRECOGNIZED_VOID_TAGNAME_MATCHER = '(?:command|embed|keygen|source|track|wbr)';
```

Comment:
My regexp knowledge is failing me here … why the question mark?

Reply:
The `?:` makes the group [non-capturing](https://www.regular-expressions.info/brackets.html), as an optimization, as we don't actually need its result stored.

Reply:
Ah, good to know. I've just learned something. Thanks for the explanation! :heart: